If no password is specified, upload firmware via TCP port  9100

dlenski opened the pull request:
Instead of uploading firmware updates to Brother printers using FTP, which
requires the admin password to be set and known, it's possible to upload them
via TCP port 9100.

This port is known as "Raw Port" in Brother printers' web-based management
interface, and elsewhere as HP JetDirect, AppSocket, or PDL-datastream.

This is the approach taken in another utility,
https://github.com/sedrubal/brother_printer_fwupd/blob/main/brother_printer_fwupd/firmware_uploader.py,
and it appears to be well-supported by Brother printers.  In particular,
this port is less likely to be disabled than FTP because "normal"
printing-related processes use it.

jcoffland commented:
Looks good but I cannot accept a PR that completely drops the old FTP method.  It would need to support both.  Also, the documentation would be out of synch.

dlenski commented:
> Looks good but I cannot accept a PR that completely drops the old FTP method. It would need to support both.

Done! Updated to use FTP if a password is specified, and port-9100 if not.

> Also, the documentation would be out of synch.

Updated!
